src/modules/src/position_controller_indi.c

```diff
@@ -0,0 +1,354 @@
+	float a33 = g13*g13 + g23*g23 + g33*g33;
+
+	// Determinant of (G'*G)
+	float detG = (a11*a22*a33 + a12*a23*a31 + a21*a32*a13) - (a13*a22*a31 + a11*a32*a23 + a12*a21*a33); 
+
+	// Inverse of (G'*G)
+	float a11_inv = (a22*a33 - a23*a32)/detG;
+	float a12_inv = (a13*a32 - a12*a33)/detG;
+	float a13_inv = (a12*a23 - a13*a22)/detG;
+	float a21_inv = (a23*a31 - a21*a33)/detG;
+	float a22_inv = (a11*a33 - a13*a31)/detG;
+	float a23_inv = (a13*a21 - a11*a23)/detG;
+	float a31_inv = (a21*a32 - a22*a31)/detG;
+	float a32_inv = (a12*a31 - a11*a32)/detG;
+	float a33_inv = (a11*a22 - a12*a21)/detG; 
+
+	// G_inv = (G'*G)_inv*G'
+	float g11_inv = a11_inv*g11 + a12_inv*g12 + a13_inv*g13;
+	float g12_inv = a11_inv*g21 + a12_inv*g22 + a13_inv*g23;
+	float g13_inv = a11_inv*g31 + a12_inv*g32 + a13_inv*g33;
+	float g21_inv = a21_inv*g11 + a22_inv*g12 + a23_inv*g13;
+	float g22_inv = a21_inv*g21 + a22_inv*g22 + a23_inv*g23;
+	float g23_inv = a21_inv*g31 + a22_inv*g32 + a23_inv*g33;
+	float g31_inv = a31_inv*g11 + a32_inv*g12 + a33_inv*g13;
+	float g32_inv = a31_inv*g21 + a32_inv*g22 + a33_inv*g23;
+	float g33_inv = a31_inv*g31 + a32_inv*g32 + a33_inv*g33;
+
+	// Lin. accel. error multiplied  G^(-1) matrix (T_tilde negated because motor accepts only positiv commands, angles are in rad)
+	indiOuter.phi_tilde   = (g11_inv*indiOuter.linear_accel_err.x + g12_inv*indiOuter.linear_accel_err.y + g13_inv*indiOuter.linear_accel_err.z);
+	indiOuter.theta_tilde = (g21_inv*indiOuter.linear_accel_err.x + g22_inv*indiOuter.linear_accel_err.y + g23_inv*indiOuter.linear_accel_err.z);
+	indiOuter.T_tilde     = -(g31_inv*indiOuter.linear_accel_err.x + g32_inv*indiOuter.linear_accel_err.y + g33_inv*indiOuter.linear_accel_err.z)/K_thr; 	
+
+	// Filter thrust
+	filter_thrust(indiOuter.thr, &indiOuter.T_incremented, &indiOuter.T_inner_f);
+
+	// Pass thrust through the model of the actuator dynamics
+	indiOuter.T_inner = indiOuter.T_inner + indiOuter.act_dyn_posINDI*(indiOuter.T_inner_f - indiOuter.T_inner); 
+
+	// Compute trust that goes into the inner loop
+		indiOuter.T_incremented = indiOuter.T_tilde + indiOuter.T_inner;
+
+	// Compute commanded attitude to the inner INDI
+	indiOuter.attitude_c.phi = indiOuter.attitude_f.phi + indiOuter.phi_tilde*180/PI;
+	indiOuter.attitude_c.theta = indiOuter.attitude_f.theta + indiOuter.theta_tilde*180/PI;	
+
+	// Clamp commands
+	indiOuter.T_incremented = clamp(indiOuter.T_incremented, MIN_THRUST, MAX_THRUST);
+	indiOuter.attitude_c.phi = clamp(indiOuter.attitude_c.phi, -10.0f, 10.0f); 	
+	indiOuter.attitude_c.theta = clamp(indiOuter.attitude_c.theta, -10.0f, 10.0f);
+
+	// Reference values, which are passed to the inner loop INDI (attitude controller)
+	refOuterINDI->x = indiOuter.attitude_c.phi;
+	refOuterINDI->y = indiOuter.attitude_c.theta;
+	// if arm is set to a nonzero value, controller increments
+	if (arm) {
+		refOuterINDI->z = indiOuter.T_incremented;
+	}
+	else {
+		refOuterINDI->z = 0;
+	} 
+
+}
+
+
+PARAM_GROUP_START(INDI_Outer)
```

Comment:
Could you call the parameter group instead: posCtrlIndi, to keep it similar with https://github.com/bitcraze/crazyflie-firmware/blob/1fdca2d2001b00e2b7f8e6501a094198f62962a1/src/modules/src/position_controller_pid.c#L284

Do you also have parameters for velocity? if not you can also just call it 'ctrlIndi'

Reply:
I am not sure if I understand this. The `PARAM`/`LOG` groups of the position INDI do not only include the param/log variables for position and velocity, but also for acceleration, thrust, gains etc.. That's why I have put all these variables in one group that belongs to the position INDI, the same way it is done in the attitude INDI file. I renamed it from `INDI_Outer` to `posCtrlIndi.`

Reply:
That is true, but we will think about the name change later. For now it is important that all the control parameters can be found at the same place as they are listed in alphabetical order in the parameter list in the cfclient. 

Thanks for changing it